codec/src/main/java/io/netty/handler/codec/protobuf/ProtobufVarint32FrameDecoder.java

```diff
@@ -60,7 +60,7 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out)
             in.resetReaderIndex();
             return;
         } else {
-            out.add(in.readBytes(length));
+            out.add(in.readSlice(length).retain());
```

Comment:
uh not even sure how this Netty issue came up on my radar but I just had a look at the date on it and probably predates adding this method 😆